### ISSUE TITLE
Toggle your goggles [see your cards as modified by effects]

### ISSUE THREAD

**Issue by kevinb9n** [terraforming-mars/terraforming-mars#2187]
Sorry I don't have drawings yet, just wanted to get this idea down for a while.

Sometimes when ogling my cards my mind boggles and I wish I could toggle on some goggles. [Imagine these toggles as separate buttons in the sidebar, or if you prefer, one button to rule them all.]

* COST goggles: click it on, and all costs of unplayed cards (#776) and standard actions (e.g. #2185) are magically rewritten based on all discount effects I have in play

* REQUIREMENT goggles: click it on, and all requirements on unplayed cards are rewritten to account for at least Inventrix/AdaptTech (discussed as part of #1407). (This might weirdly rewrite to "max 9 oceans" or maybe even "-32 C" but won't hide the requirement completely, because Cutting-Edge.)

* COUNT THE THINGS goggles: click it on, and all instructions on unplayed cards that involve either the "per" icon like Medical Lab, or a conditional OR clause like Nitro-Rich, are rewritten to show the actual amount it would currently give. Also rewrites card-actions I have in play like Orbital Cleanup, and VPs like on Water Import. (To be sure, you would often like to know the raw measurement in question, or at least have a way to know whether you are on the cusp of the next bump up, but that would require actual UI of some kind instead of just rewriting using existing UI we already have. It can be considered independently.)

* TRIGGER goggles: (putting these here only to further illustrate the general goggle concept, but I have no doubt that it's too hard) click it on, and see on each card (somehow) additional instructions from your active "when-play-card/tag" triggers. Example: after playing OptAero, now see 3 MC 3 Heat on all your space events. (Warning: there is very much a slippery slope here, and it will never be feasible to really show every single thing clearly that will result.)

**Important clarifications**

* Each goggle(?) has to be *very easy* to turn on and off, without having to go into settings, or this whole thing will suck
* To keep cards simple and clean, the goggle concept is 100% about rewriting a card to look "exactly" like a different card that *could* itself have been printed - except:
* It's crucial to always have *some* visual indicator that the information has been changed. For example, change text from black to either dark green (persistent effect) or dark red (includes at least one temporary effect, like from Special Design)
* Goggles shouldn't change card text. So even with goggles on, I can still compare the raw (text) and cooked (graphic) forms.
* While a desktop user mouse-hovers on a cost/requirement/etc, it could flip to the opposite goggle mode temporarily.
* The more we shift to showing standard projects *and even standard actions* in the style of https://github.com/bafolts/terraforming-mars/issues/2185#issuecomment-751271972, the more that players of cards like Ecoline, Thorgate, Cryo Sleep/Rim Freighters will benefit from the goggles concept.

**Comment by kevinb9n** [comment]
I forgot to address the "why".

We can divide the mental activity of gaming broadly into at least two processes: (1) accurately assessing facts about the game state (including correctly predicting the immediate consequences of a potential decision); and (2) decision-making based on those facts. If you do excellent decision-making but missed a key fact, then you may make a very bad play.

To many players, both are just part of the game (and, they feel, should be). Partly because if they ever play competitively IRL then they just don't have much choice. They're gonna need those skills. Yes, we have tag trackers and milestone trackers and such, but these are justified simply by how *time-consuming* it is to keep counting up those values by hand.

But for the rest of us, well. I think mental sport (1) is the boring one and mental sport (2) is the interesting one. The more that (1) is helped by the UI, then the more the players can play game (2) instead. And I would much rather play against my opponent's best *intentional* plays than against blunders they made because of (1). This is also exactly why I strongly prefer to enable undo. All of this might be only because I'm a casual, but I don't know.

Long story short: it's possible that eventually some game interface features would want to sit behind a "casual mode" game option.

**Comment by kevinb9n** [comment]
So we have discounts now. Yay!

When time permits let's consider a switch to the toggle approach.

Quick summary of the idea: a button in the sidebar changes between always showing printed costs or discounted costs. A discounted cost appears in green digits instead of black. On desktop, hover to see the cost from the opposite mode.

Some advantages:

* I think card space is very precious in general. I suspect they look quite crowded to new players in particular.
* We want to be able to do this for requirements (e.g. Inventrix) too - even more space.
* The actual need to know the printed cost comes up only occasionally (Advertising, etc.) When it does, you don't really need to see both costs at *once*. Just a way to see either. 
* Experienced players know all the printed costs anyway, so this lets them get out of the way.
* When showing both costs as it does now, players don't seem to share a consensus on which cost should be which.

**Comment by kevinb9n** [comment]
It was an idea but didn't go anywhere and that's fine.
